IntegrationWithPharmacies/Controllers/ReportController.cs

```diff
@@ -28,25 +31,91 @@ public ReportController(MyDbContext context)
             StringBuilder stringBuilder = new StringBuilder();
 
+            return writeReportSftp(date, sftpService, complete, stringBuilder);
+        }
+
+        private IActionResult writeReportSftp(DateOfOrder date, SftpService sftpService, string complete, StringBuilder stringBuilder)
+        {
             foreach (RegistrationInPharmacy registration in RegistrationInPharmacyService.GetAll())
             {
                 stringBuilder.Append(registration.ApiKey + ";");
             }
-            System.IO.File.WriteAllText(testFile, stringBuilder.ToString() + "!    Report about consumption of medicine\n\n\n" +getReportText(date));
-            sftpService.UploadFile(testFile, @"\pub\" + Path.GetFileName(testFile));
+            System.IO.File.WriteAllText(complete, stringBuilder.ToString() + "!    Report about consumption of medicine\n\n\n" + getReportText(date));
+            sftpService.UploadFile(complete, @"\pub\" + complete);
+            SendNotificationAboutReport();
             return Ok();
         }
 
+        public int getRandomNumber()
+        {
+            return new Random().Next(1, 100);
+        }
+        [HttpPost("http")]
+        public IActionResult PostHttp(DateOfOrder date)
+        {
+            if (Environment.Equals("Development"))
+            {
+                return writeInFileHttp(date);
+            }
+            return BadRequest();
+        }
+
+        private IActionResult writeInFileHttp(DateOfOrder date)
+        {
+            var testFile = @"..\test.txt";
+            String complete = @"FileReports\..\Report_" + DateTime.Now.ToString("dd-MM-yyyy") + "_" + getRandomNumber() + ".txt";
+            StringBuilder stringBuilder = new StringBuilder();
+
+            foreach (RegistrationInPharmacy registration in RegistrationInPharmacyService.GetAll())
+            {
+                stringBuilder.Append(registration.ApiKey + ";");
+            }
+            return writeReportHttp(date, testFile, complete, stringBuilder);
+        }
+
+        private IActionResult writeReportHttp(DateOfOrder date, string testFile, string complete, StringBuilder stringBuilder)
+        {
+            System.IO.FileStream fs = System.IO.File.Create(complete);
+            fs.Close();
+            System.IO.File.WriteAllText(complete, stringBuilder.ToString() + "!    Report about consumption of medicine\n\n\n" + getReportText(date));
+            try
+            {
+                uploadFile(complete);
+                return Ok(JsonConvert.SerializeObject(testFile));
+            }
+            catch (Exception e) { return BadRequest(); }
+        }
+
+        public void uploadFile(String complete)
+        {
+               WebClient client = new WebClient();
+               Uri uri = new Uri(@"http://localhost:8082/download/file/http");
+               client.Credentials = CredentialCache.DefaultCredentials;
+               client.UploadFile(uri, "POST", complete);

```

Comment:
Write this in form:
" client.UploadFile(new Uri(@"http://localhost:8082/download/file/http"), "POST", complete);"